app/routes/_auth.signup.tsx

```diff
@@ -1,45 +1,64 @@
 import type { LoaderArgs, ActionArgs } from '@remix-run/node';
 import { redirect } from '@remix-run/node';
 import { Form, Link, useActionData, useNavigation } from '@remix-run/react';
+import { z } from 'zod';
+
 import { Card } from '~/components/ui/containers';
 import { Input } from '~/components/ui/forms';
 import { getHash } from '~/modules/database/crypto.server';
 import { db } from '~/modules/database/db.server';
+import { badRequest } from '~/modules/response/response.server';
 import { createUserSession, getUserSession } from '~/modules/session/session.server';
 
 export async function action({ request }: ActionArgs) {
-  const form = await request.formData();
-  const name = form.get('name');
-  const email = form.get('email');
-  const password = form.get('password');
-  const confirmPassword = form.get('confirmPassword');
-
-  if (!name || !email || !password || !confirmPassword) {
-    return { status: 400, message: 'Missing required fields' };
-  }
+  const signUpFormData = Object.fromEntries(await request.formData());
 
-  if (password !== confirmPassword) {
-    return { status: 400, message: 'Passwords do not match' };
-  }
+  const SignUpForm = z.object({
```

Comment:
Why uppercasing? Zod convention?

Reply:
Yes, the Zod documentation seems to use uppercase when defining `z.object()` and `z.eum()` schemas and lowercase when defining primitives (i.e. `z.string()`, ` z.number()`, etc.). Casing does not matter functionally though so we should go with whichever casing the team prefers! 

During our call, it was decided to use lowercase, so let's stick with that in the future :)